File: csharp/Svix/BackgroundTask.cs
using System;
using System.Collections.Generic;
using System.Net;
using System.Threading;
using System.Threading.Tasks;
using Microsoft.Extensions.Logging;
using Xwebhook.Abstractions;
using Xwebhook.Api;
using Xwebhook.Client;
using Xwebhook.Model;
using Xwebhook.Models;

namespace Xwebhook
{
    public sealed class BackgroundTask : XwebhookResourceBase, IBackgroundTask
    {
        private readonly IBackgroundTasksApi _backgroundTaskApi;

        public BackgroundTask(IXwebhookClient xwebhookClient, IBackgroundTasksApi backgroundTaskApi)
            : base(xwebhookClient)
        {
            _backgroundTaskApi = backgroundTaskApi ?? throw new ArgumentNullException(nameof(backgroundTaskApi));
        }

        public BackgroundTaskOut Get(string taskId, string idempotencyKey = default)
        {
            try
            {
                var lBackgroundTask = _backgroundTaskApi.GetBackgroundTask(taskId);

                return lBackgroundTask;
            }
            catch (ApiException e)
            {
                Logger?.LogError(e, $"{nameof(Get)} failed");

                if (Throw)
                    throw;

                return null;
            }
        }

        public async Task<BackgroundTaskOut> GetAsync(string taskId, string idempotencyKey = default, CancellationToken cancellationToken = default)
        {
            try
            {
                var lBackgroundTask = await _backgroundTaskApi.GetBackgroundTaskAsync(taskId);

                return lBackgroundTask;
            }
            catch (ApiException e)
            {
                Logger?.LogError(e, $"{nameof(GetAsync)} failed");

                if (Throw)
                    throw;

                return null;
            }
        }

        public ListResponseBackgroundTaskOut List(BackgroundTaskListOptions options = null, string idempotencyKey = default)
        {
            try
            {
                var lResponse = _backgroundTaskApi.ListBackgroundTasks(
                    options?.Status,
                    options?.Task,
                    options?.Limit,
                    options?.Iterator,
                    options?.Order);

                return lResponse;
            }
            catch (ApiException e)
            {
                Logger?.LogError(e, $"{nameof(List)} failed");

                if (Throw)
                    throw;

                return new ListResponseBackgroundTaskOut();
            }
        }

        public async Task<ListResponseBackgroundTaskOut> ListAsync(BackgroundTaskListOptions options = null, string idempotencyKey = default, CancellationToken cancellationToken = default)
        {
            try
            {
                var lResponse = await _backgroundTaskApi.ListBackgroundTasksAsync(
                    options?.Status,
                    options?.Task,
                    options?.Limit,
                    options?.Iterator,
                    options?.Order,
                    cancellationToken);

                return lResponse;
            }
            catch (ApiException e)
            {
                Logger?.LogError(e, $"{nameof(ListAsync)} failed");

                if (Throw)
                    throw;

                return new ListResponseBackgroundTaskOut();
            }
        }
    }
}

File: java/lib/src/main/java/com/svix/BackgroundTask.java
package com.xwebhook;

import com.xwebhook.exceptions.ApiException;
import com.xwebhook.internal.api.BackgroundTasksApi;
import com.xwebhook.models.BackgroundTaskOut;
import com.xwebhook.models.ListResponseBackgroundTaskOut;

public final class BackgroundTask {
    private final BackgroundTasksApi api;

    BackgroundTask() {
        api = new BackgroundTasksApi();
    }

    public ListResponseBackgroundTaskOut list(final BackgroundTaskListOptions options) throws ApiException {
        try {
            return api.listBackgroundTasks(options.getStatus(), options.getTask(), options.getLimit(),
                    options.getIterator(),
                    options.getOrder());
        } catch (com.xwebhook.internal.ApiException e) {
            throw Utils.wrapInternalApiException(e);
        }
    }

    public BackgroundTaskOut get(final String taskId) throws ApiException {
        try {
            return api.getBackgroundTask(taskId);
        } catch (com.xwebhook.internal.ApiException e) {
            throw Utils.wrapInternalApiException(e);
        }
    }
}
